lib/lure/npm.go

```diff
@@ -85,7 +84,22 @@ func updateJSON(parsedPackageJSON *packageJSON, key string, module string, versi
 	_, ok := (*parsedPackageJSON)[key]
 	if ok {
 		dependencies := (*parsedPackageJSON)[key].(map[string]interface{})
-		dependencies[module] = version
-		(*parsedPackageJSON)[key] = dependencies
+
+		// https://docs.npmjs.com/misc/semver#x-ranges-12x-1x-12-
+		r, _ := regexp.Compile("^(\\^|~).*")
+
+		// Only update the dependency if it exists already.
+		if dependencies[module] != nil {
+			// Check for version operators and reuse them
+			operators := r.FindStringSubmatch(dependencies[module].(string))
```

Comment:
I think it would help readability if there was a "hasRangeOperator(version)" function that would regroup this check.